docs/developer/testing.md

```diff
@@ -0,0 +1,188 @@
+
+```bash
+lt --port 8000
+```
+
++ You will receive a URL, for example https://gqgh.localtunnel.me, that you can share with anyone for as long as your local instance remains active. Any requests will be routed to your local service at the specified port.
+
+### With Sauce Labs
+
+[Sauce Labs](https://saucelabs.com/beta/dashboard/builds) allows collaborators and contributors to execute automated tests against Selenium WebDriver on pre-configured browser matrix's.
+
+The cross browser testing strategy focuses on the differences between browser rendering engines. There are three different [configuration files](https://github.com/Microsoft/fast-dna/tree/master/build/testing) based on feature maturity of ongoing work.
+
+[Alpha](https://github.com/Microsoft/fast-dna/blob/master/build/testing/config-browsers.alpha.js) Immature feature development
+
+```javascript
+node build/testing/sauce-labs/test-browsers.js alpha
+```
+
+[Beta](https://github.com/Microsoft/fast-dna/blob/master/build/testing/config-browsers.beta.js) Stable feature development
+
+```javascript
+node build/testing/sauce-labs/test-browsers.js beta
+```
+
+[Release](https://github.com/Microsoft/fast-dna/blob/master/build/testing/config-browsers.release.js) Mature feature development
+
+```javascript
+node build/testing/sauce-labs/test-browsers.js release
+```
+
+### With Applitools
+
+[Applitools](https://eyes.applitools.com/) can be used to visual regression test UI changes on your own branches allow  to execute automated tests against Selenium WebDriver. Currently, ChromeDriver and FirefoxDriver are supported.
+
+Prior to beginning to make UI changes on a new branch it's recommended to run initially to create a baseline set of images.
+
+1. The first time the test is executed on a new branch, screenshots are created as 'undefined' status. 
+2. After the tests finish users should select all checkbox's and choose to save to generate the baseline image.
+3. Next perform UI code changes
+4. Rerun the first set of tests and for all snapshots that match the baseline they will turn green and show as 'passed' status. Any 'undefined' status would be additional UI changes added. Though to get into this scenario new test cases would be required in the `./run-msft-docs.js` file.  Any anomalies will be marked red as 'failed' status.
+5. Manually review all UI changes and if the changes are desired select and choose to save the new baseline. Anything, else should be changed and retested.
+
+```javascript
+node build/testing/applitools/test-javascript.js
+```
+
+#### Visual Regression testing with Cypress SDK
+
+Contributors can perform functional user interface and experience testing using [Cypress](https://www.cypress.io/). This does currently include visual regression testing, though, a feature that has been requested. Cypress is not based on Selenium WebDriver and instead is built completely on JavaScript enabling significant performance advantages.
```

Comment:
This line is confusing "This does currently include visual regression testing, though, a feature that has been requested" does it or does it not have this feature?

Reply:
I don't think we need to talk about what's isn't available so I removed.